esphome/components/nfc/nfc.cpp

```diff
@@ -0,0 +1,89 @@
+#include "nfc.h"
+#include "esphome/core/log.h"
+
+namespace esphome {
+namespace nfc {
+
+static const char *TAG = "nfc";
+
+std::string format_uid(std::vector<uint8_t> uid) {
```

Comment:
```suggestion
std::string format_uid(std::vector<uint8_t> uid) {
```

also, better to use `const std::vector<uint8_t> &uid` here, otherwise the whole vector is copied each time this method is called (applies to methods below as well)